common/src/main/java/io/netty/util/concurrent/NonStickyEventExecutorGroup.java

```diff
@@ -0,0 +1,308 @@
+
+    @Override
+    public ScheduledFuture<?> schedule(Runnable command, long delay, TimeUnit unit) {
+        return group.schedule(command, delay, unit);
+    }
+
+    @Override
+    public <V> ScheduledFuture<V> schedule(Callable<V> callable, long delay, TimeUnit unit) {
+        return group.schedule(callable, delay, unit);
+    }
+
+    @Override
+    public ScheduledFuture<?> scheduleAtFixedRate(Runnable command, long initialDelay, long period, TimeUnit unit) {
+        return group.scheduleAtFixedRate(command, initialDelay, period, unit);
+    }
+
+    @Override
+    public ScheduledFuture<?> scheduleWithFixedDelay(Runnable command, long initialDelay, long delay, TimeUnit unit) {
+        return group.scheduleWithFixedDelay(command, initialDelay, delay, unit);
+    }
+
+    @Override
+    public boolean isShutdown() {
+        return group.isShutdown();
+    }
+
+    @Override
+    public boolean isTerminated() {
+        return group.isTerminated();
+    }
+
+    @Override
+    public boolean awaitTermination(long timeout, TimeUnit unit) throws InterruptedException {
+        return group.awaitTermination(timeout, unit);
+    }
+
+    @Override
+    public <T> List<java.util.concurrent.Future<T>> invokeAll(
+            Collection<? extends Callable<T>> tasks) throws InterruptedException {
+        return group.invokeAll(tasks);
+    }
+
+    @Override
+    public <T> List<java.util.concurrent.Future<T>> invokeAll(
+            Collection<? extends Callable<T>> tasks, long timeout, TimeUnit unit) throws InterruptedException {
+        return group.invokeAll(tasks, timeout, unit);
+    }
+
+    @Override
+    public <T> T invokeAny(Collection<? extends Callable<T>> tasks) throws InterruptedException, ExecutionException {
+        return group.invokeAny(tasks);
+    }
+
+    @Override
+    public <T> T invokeAny(Collection<? extends Callable<T>> tasks, long timeout, TimeUnit unit)
+            throws InterruptedException, ExecutionException, TimeoutException {
+        return group.invokeAny(tasks, timeout, unit);
+    }
+
+    @Override
+    public void execute(Runnable command) {
+        group.execute(command);
+    }
+
+    private static final class NonStickyOrderedEventExecutor extends AbstractEventExecutor
+            implements Runnable, OrderedEventExecutor {
+        private final EventExecutor executor;
+        private final Queue<Runnable> tasks = PlatformDependent.newMpscQueue();
+
+        private static final int NONE = 0;
+        private static final int SUBMITTED = 1;
+        private static final int RUNNING = 2;
+
+        private final AtomicInteger state = new AtomicInteger();
+
+        NonStickyOrderedEventExecutor(EventExecutor executor) {
+            super(executor);
+            this.executor = executor;
+        }
+
+        @Override
+        public void run() {
+            if (!state.compareAndSet(SUBMITTED, RUNNING)) {
+                return;
+            }
+            try {
+                for (;;) {
```

Comment:
let me know if I can help ... are you thinking we may need a solution more complicated than the following?

```java
try {
  int i = 0;
  for (; i < maxRunCount; ++i) {
    ..
  }
} finally {
  state.set(NONE);
  if (i == maxRunCount) {
    executor.execute(this); // This should share code with the Execute method below and cover the same exception cases, etc...
  }
}
```